src/cudadecoder/cuda-online-pipeline-dynamic-batcher.cc

```diff
@@ -167,5 +179,10 @@ void CudaOnlinePipelineDynamicBatcher::WaitForCompletion() {
   cuda_pipeline_.WaitForLatticeCallbacks();
 }
 
+int CudaOnlinePipelineDynamicBatcher::GetPendingChunks(CorrelationID corr_id) {
+  std::lock_guard<std::mutex> lk(next_batch_and_backlog_m_);
+  return n_chunks_per_corr_[corr_id];
```

Comment:
`n_chunks_per_corr_.at(corr_id);` is better  as it checks whether the given key exists or not.

This method is a getter I think, so you should check if the key exists or not.

Reply:
Done. Actually there is a catch here. If corr_id is finalized, we still need to return 0. I've added comment about it.